src/Lib.hs

```diff
@@ -0,0 +1,191 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Lib (app) where
+
+import Control.Monad
+import Control.Monad.IO.Class
+import Control.Monad.Trans.Except (ExceptT (..), runExceptT, withExceptT)
+import Data.ByteString (ByteString)
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy.Char8 as BSL
+import Data.Coerce (coerce)
+import Data.Either (isLeft)
+import Data.Maybe (fromMaybe)
+import Database.SQLite.Simple
+import Database.SQLite.Simple.FromField (FromField (..))
+import Database.SQLite.Simple.ToField (ToField (..))
+import GHC.IO.Exception (ExitCode (ExitSuccess))
+import Network.Wai.Handler.Warp (run)
+import Servant
+import qualified System.Directory as Dir
+import System.FilePath ((</>))
+import qualified System.Process as Proc
+
+-- | Top-level API. Examples:
+--   localhost:8081/ps
+--   localhost:8081/procps/ps?sys=aarch64-linux
+--  TODO API documentation
+type API =
+  Get '[JSON] NoContent
+    :<|> Capture "binary-name" BinaryName :> QueryParam "sys" System :> Get '[OctetStream] ByteString
+    :<|> Capture "package-name" PackageName :> Capture "binary-name" BinaryName :> QueryParam "sys" System :> Get '[OctetStream] ByteString
+
+newtype BinaryName = BinaryName {unBinaryName :: String}
+  deriving newtype (ToField, FromHttpApiData)
+
+newtype PackageName = PackageName {unPackageName :: String}
+  deriving newtype (Eq, Ord, FromField, FromHttpApiData, ToField)
+
+-- TODO just make this a string?
+-- TODO isn't this an architecture more than a system? nix calls it a system I guess
```

Comment:
Then shall we name it `ArchSystem` or something?
Just `System` looks potentially confusing to me.

Reply:
>  Just System looks potentially confusing to me.

100% agreed, although as far as I can tell it's the most common/appropriate name for the architecture + OS slug...

I'm not sure ArchSystem is much better though, although I guess it _is_ an arch + a system

Reply:
Yeah, I was meaning an arch + a system.

Reply:
It's very close to what GNU calls a [triplet](https://wiki.osdev.org/Target_Triplet) :speak_no_evil:

Reply:
Oh, triplet 👀 
Then why don't we call ours `TargetDoublet`?

Or how about `Platform`?
The page explains it as a combination of three "platform"s: "build platform", "host platform" and "target platform".

Reply:
`Platform` is good!